include/cuco/detail/traits.hpp

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ */
+
+#pragma once
+
+#include <thrust/device_reference.h>
+#include <thrust/tuple.h>
+
+#include <tuple>
+#include <type_traits>
```

Comment:
Since this is a new file, we should switch to the `cuda/std` namespace when possible to get rid of the `--relaxed-constexpr` flag at some point.

Reply:
good call. moved to libcuda++ when possible.